### PR TITLE
feat: buildFindMany

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

a function to build "findMany" functions for database Models, simalar to `buildInsertInto`.

## Testing

`find-many.test.ts`

Notice that, instead of mocking and testing database return data, only compare generated sql string and expected sql string.

### CODE REVIEW

File: packages/core/src/database/find-many.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const findMany = buildFindMany(pool, Users);
+    await findMany();
+  });
+
+  it('matches expected sql with where conditions', async () => {
+    const pool = createTestPool(
+      'select "id", "username", "primary_email", "primary_phone", "password_encrypted", "password_encryption_method", "password_encryption_salt"\nfrom "users"\nwhere "id"=$1'
+    );
+    const findMany = buildFindMany(pool, Users);
+    await findMany({ where: { id: '123' } });
```
Comment (simeng-li) [inline packages/core/src/database/find-many.test.ts:21]: Feels not able to test the findMany returned query script in this way.  
Could mock the returning value using `createTestPool` and test the returning value. Of expect not to throw any exceptions?

Comment (simeng-li) [reply packages/core/src/database/find-many.test.ts:21]: We may come up with a better query test practice later.

Comment (wangsijie) [reply packages/core/src/database/find-many.test.ts:21]: discussed offline, `buildFindMany` created a deferred function that will be invoked in `findMany`

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**